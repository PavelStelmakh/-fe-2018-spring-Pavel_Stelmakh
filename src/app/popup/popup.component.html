<div [class.showPopup]="show">
  <ng-content></ng-content>  
</div>
