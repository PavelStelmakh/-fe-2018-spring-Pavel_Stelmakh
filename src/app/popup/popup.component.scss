:host {
    display: block;
    div {
        position: absolute;
        width: 300px;
        height: 70px;
        background-color: rgba(102, 102, 102, 0.8);
        border: 1px solid #666;
        border-left-width: 10px;
        top: -72px;
        left: calc(50vw - 150px);
        transition: top 500ms linear;
        line-height: 70px;
        font-family: Verdana;
        text-align: center;
        color: #ccc;
        font-size: 1.1rem;
    }

    .showPopup {
        top: 0;
    }

}
