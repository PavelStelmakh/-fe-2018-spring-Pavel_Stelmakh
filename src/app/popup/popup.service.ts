import { Injectable, OnDestroy } from '@angular/core';
import { BehaviorSubject } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PopupService implements OnDestroy {
  showPopup: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);

  constructor() { }

  show() {
    this.showPopup.next(true);
   setTimeout(() => this.showPopup.next(false), 3000);
  }

  ngOnDestroy() {
    this.showPopup.complete();
  }

}
