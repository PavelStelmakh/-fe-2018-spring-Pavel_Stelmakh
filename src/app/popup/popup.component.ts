import { Component, OnInit } from '@angular/core';
import { PopupService } from './popup.service';

@Component({
  selector: 'app-popup',
  templateUrl: './popup.component.html',
  styleUrls: ['./popup.component.scss']
})
export class PopupComponent implements OnInit {
  show: boolean;

  constructor(private popup: PopupService) {}

  ngOnInit() {
    this.show = false;
    this.popup.showPopup.subscribe((result: boolean) => this.show = result);
  }

}
