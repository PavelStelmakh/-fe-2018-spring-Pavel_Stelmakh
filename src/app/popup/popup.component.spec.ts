import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { PopupComponent } from './popup.component';
import { PopupService } from './popup.service';

describe('PopupComponent', () => {
  let fixture: ComponentFixture<PopupComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        PopupComponent
      ],
      providers: [
        PopupService
      ],
      schemas: [
       NO_ERRORS_SCHEMA
      ]
    }).compileComponents();

  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PopupComponent);
  });

  it('renders markup to snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

});
