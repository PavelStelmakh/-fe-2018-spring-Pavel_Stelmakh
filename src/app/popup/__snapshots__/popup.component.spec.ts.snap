// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupComponent renders markup to snapshot 1`] = `
<app-popup
  popup={[Function PopupService]}
>
  <div />
</app-popup>
`;
